packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -320,10 +330,56 @@ markdown based on [CommonMark](http://commonmark.org/) flavor:
 -   Italics - `*Italics*`
 -   Numbered list - `1. Numbered list`
 -   Bulleted list - `* Bulleted list`
+-   Absolute URL links - `<Absolute URI link>` (For more details on the markdown syntax for absolute URL links, see [Autolinks in CommonMark](https://spec.commonmark.org/0.30/#autolink))
+
+_Configurations on Tiptap to support only absolute links_:
+
+Install the [link extension](https://tiptap.dev/api/marks/link) mark from Tiptap and initialize the `Links` with the following configurations:
+
+1.  Set regular expression in [validate](https://tiptap.dev/api/marks/link#validate) field to support only `HTTP` and `HTTPS` absolute links in the editor.
```

Comment:
Still think that any validation being done client side needs to be replicated server side. For example that the server does the same validation (markdown only with certain features is accepted, must validate with the same schema, must have links with certain schemas. etc). How / where is that being tracked? As more features are added etc how would they stay aligned?

Reply:
@rajsite, currently in the backend, we do not plan to add any validation on the markdown content. Because even if try to create a comment (that includes links that do not belong to `http` or `https` schema) directly via API, when it gets rendered in the UI, it will be rendered as plain text (because of the rules we configured during the markdown parser).

Currently, SLE email notification does not support HTML content. In the future, when it supports HTML content, we can implement the same parsing logic (we are doing in the client to convert markdown to HTML) to generate the HTML content for mailing.

Please let me know your thoughts.

Reply:
As long as it's crystal clear on the backend owner's part that no validation of the data going into the backend is happening (i.e. only client side validation which is a known bad practice) and it passes muster for their security audits. I.e. those decisions have sign-off from the right folks understanding and documenting those concerns, ie. @rbell517

I think it's weird that SystemLink's security requirements are being encoded in the nimble rich text editor component this way and that it's the only line of validation that happens. 

Ultimately I won't block this spec on it. The behavior being implemented for the rich-text components as described here seems fine (only autolinking http and https urls).

Reply:
Yes this security question has come up a few times, and essentially where I have landed is we should always consider the contents of a comment as untrusted, in both the frontend and any usage of it in the backend. Just as the backend shouldn't assume its getting contents from a specific frontend, the frontend shouldn't assume its getting content from a specific backend; these should be trust boundaries for both. 

That said, even if the backend treats the comment contents as untrusted, that does not mean it will treat it the same as untrusted content in the frontend. Its not rendering the content, but it is inserting it into a database, so the security context changes to validating for SQL injections instead of XSS, for example. This will change again for other usages like email notifications, etc. Hopefully the posture is clear enough.

Reply:
> This will change again for other usages like email notifications

So if I understand as of now that means we would accept arbitrary strings even if they don't conform to markdown encoded in our supported subset of CommonMark features and custom extensions (i.e. mentions syntax and however that will look in the future) in addition to XSS concerns being detected in properly structured formats.

That seems like something that would be challenging to clean-up without data loss in the future when we could make sure to start enforcing that now. I'd really hope we don't introduce that technical debt into the system if we don't need to but yea ultimately as stated that's not a nimble concern and sounds like the concerns are understood.

I guess the remaining question is if that is tracked anywhere @rbell517 @suseendran-ni  (security concerns for string insertion and technical debt of markdown validation so we know what's actually being dumped in our databases), but that is orthogonal to acceptance of this PR (but wouldn't mind seeing how that is captured).

Reply:
Created a tech debt to track this conversation
https://dev.azure.com/ni/DevCentral/_workitems/edit/2511073

@vivinkrishna-ni @vikisekarNI @rbell517 @rajsite